tools/buildsteps/windows/vswhere.bat

```diff
@@ -1,5 +1,19 @@
 @ECHO OFF
 
+IF "%1"=="" (
+  ECHO ERROR! vswhere.bat: architecture not specified
+  EXIT /B 1
+)
+
+REM running vcvars more than once can cause problems; exit early if using the same configuration, error if different
+IF "%VSWHERE_SET%"=="%*" (
+  ECHO vswhere.bat: VC vars already configured for %VSWHERE_SET%
+  GOTO :EOF
+)
+IF "%VSWHERE_SET%" NEQ "" (
+  ECHO ERROR! vswhere.bat: VC vars are configured for %VSWHERE_SET%
+  EXIT /B 1
+)
```

Comment:
Why not allow switching configuration?
I know it may break, but if you you just want to compile FFmpeg for multiple platforms from a cmd you now can't anymore.

Reply:
Mostly because it's undefined behavior and too many switches will result in a corrupted environment (granted you can escape from it by restarting the CMD window).

When it breaks, though, it's cryptic:

    The input line is too long.
    The syntax of the command is incorrect.

For instance, if I run `C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\vc\Auxiliary\Build\vcvars64.bat` 4 times I'll hit this. If I alternate between 64-bit and 32-bit it'll hit this as well on the 5th run and the Path will be over 8000 characters long. If you do `echo.%PATH%01234...` it starts giving `The input line is too long.` after ~150 more characters.

See [this](https://stackoverflow.com/questions/21723275/how-to-override-batch-variable-number-of-characters-limit) SO question.

Running the command `cmd /k x86\make-mingwlibs.bat` from `tools\buildsteps\windows` will allow you to preserve your command window. In my other dev environments the build aliases typically call "cmd /c ..." when calling a build script to avoid polluting the environment.